pkg/controller/ingress/ingress_controller_test.go

```diff
@@ -54,6 +55,60 @@ var (
 	}
 )
 
+func TestRouteMigration(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+
+	test := struct {
+		name        string
+		exRoute     *routev1.Route
+		wantName    string
+		removedName string
+	}{
+		name: "Clean up old route and new route is generated",
+		exRoute: &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-0"},
```

Comment:
should this have the correct labels set? Could we add a second route here that has different labels and make sure we only remove the one that matches?

Reply:
> Could we add a second route here that has different labels and make sure we only remove the one that matches?

Yes, we should add it. But it looks like __fake__ client in current K-O-I does not support labels list option added by https://github.com/kubernetes-sigs/controller-runtime/commit/e9c5505ea8a3bf1bb77b5f15a9fd904c597e63e6, so we cannot test it for now.
We need to upgrade the controller-runtime version for the test.

Reply:
ugh. Can we safely do that? A separate PR first maybe?

Reply:
I did check the possibility of doing so quickly:
- @nak3's referenced commit is part of controller-runtime since v0.2.0.
- To get that, we'd need to upgrade to osdk v0.11.0
- osdk v0.11.0 also bumped k8s clients to `kubernetes-1.14.1` though, but knative-serving wants `kubernetes-1.13.1`.

It might be more trouble than it's worth it at this point. I'd personally be fine if we just check this manually for this release and make sure it does the right thing. Alternatively, could we implement the label matching in our code, leaving a comment that we do this in addition to the List filter so that we can test things?